pkg/controller/oneagent/istio.go

```diff
@@ -17,105 +17,152 @@ import (
+	add, err := r.reconcileIstioCreateConfigurations(instance, comHosts, role, logger)
 	if err != nil {
-		logger.Error(err, "error reconciling Istio config")
-		return err
+		logger.Error(err, "istio: failed to create objects")
+		return false, err
 	}
 
-	r.reconcileIstioRemoveConfigurations(instance, comHosts, logger)
+	rem, err := r.reconcileIstioRemoveConfigurations(instance, comHosts, role, logger)
+	if err != nil {
+		logger.Error(err, "istio: failed to delete objects")
+		return false, err
+	}
 
-	return nil
+	return add || rem, nil
 }
 
 func (r *ReconcileOneAgent) reconcileIstioRemoveConfigurations(instance *dynatracev1alpha1.OneAgent,
-	comHosts []dtclient.CommunicationHost, logger logr.Logger) {
+	comHosts []dtclient.CommunicationHost, role string, logger logr.Logger) (bool, error) {
 
 	listOps := &client.ListOptions{
 		Namespace:     instance.Namespace,
-		LabelSelector: labels.SelectorFromSet(buildLabels(instance.Name)),
+		LabelSelector: labels.SelectorFromSet(buildIstioLabels(instance.Name, role)),
 	}
 
 	seen := map[string]bool{}
 	for _, ch := range comHosts {
 		seen[istio.BuildNameForEndpoint(instance.Name, ch.Host, ch.Port)] = true
 	}
 
-	r.reconcileIstioRemoveConfiguration(instance, istio.VirtualServiceGVK, listOps, seen, logger)
-	r.reconcileIstioRemoveConfiguration(instance, istio.ServiceEntryGVK, listOps, seen, logger)
+	vsUpd, err := r.reconcileIstioRemoveConfiguration(instance, istio.VirtualServiceGVK, listOps, seen, logger)
+	if err != nil {
+		return false, err
+	}
+
+	seUpd, err := r.reconcileIstioRemoveConfiguration(instance, istio.ServiceEntryGVK, listOps, seen, logger)
+	if err != nil {
+		return false, err
+	}
+
+	return vsUpd || seUpd, nil
 }
 
 func (r *ReconcileOneAgent) reconcileIstioRemoveConfiguration(instance *dynatracev1alpha1.OneAgent, gvk schema.GroupVersionKind,
-	listOps *client.ListOptions, seen map[string]bool, logger logr.Logger) {
+	listOps *client.ListOptions, seen map[string]bool, logger logr.Logger) (bool, error) {
 
 	var list unstructured.UnstructuredList
 	list.SetGroupVersionKind(gvk)
 
 	if err := r.client.List(context.TODO(), listOps, &list); err != nil {
-		return
+		return false, err
 	}
 
+	upd := false
+
 	for _, item := range list.Items {
 		if _, ok := seen[item.GetName()]; !ok {
+			upd = true
 			logger.Info(fmt.Sprintf("removing Istio %s: %v", gvk.Kind, item.GetName()))
 			if err := r.client.Delete(context.TODO(), &item); err != nil {
-				logger.Info(fmt.Sprintf("failed to delete Istio %s: %v", gvk.Kind, err))
-				continue
+				return false, fmt.Errorf("failed to delete Istio %s: %v", gvk.Kind, err)
 			}
 		}
 	}
+
+	return upd, nil
 }
 
 func (r *ReconcileOneAgent) reconcileIstioCreateConfigurations(instance *dynatracev1alpha1.OneAgent,
-	comHosts []dtclient.CommunicationHost, logger logr.Logger) error {
+	comHosts []dtclient.CommunicationHost, role string, logger logr.Logger) (bool, error) {
+
+	upd := false
```

Comment:
Would it make sense to name this var also differently? Like `created` or similar...